YMChat/src/main/java/com/yellowmessenger/ymchat/BotWebView.java

```diff
@@ -202,6 +203,19 @@ public void onCreate(Bundle savedInstanceState) {
             this.finish();
         });
         showCloseButton();
+
+        RelativeLayout constraintLayout = findViewById(R.id.parentView);
+        constraintLayout.getViewTreeObserver().addOnGlobalLayoutListener(() -> {
+            Rect r = new Rect();
+            constraintLayout.getWindowVisibleDisplayFrame(r);
+            int screenHeight = constraintLayout.getRootView().getHeight();
+            int keypadHeight = screenHeight - r.bottom;
+            if (keypadHeight > screenHeight * 0.15) {
```

Comment:
can we add values in dimens.xml?

Reply:
No, dimens is used to provide dimension such as margin, padding , spacing etc and has values in dp, dip, and px. here we need a float value and it's completely local and not being used anywhere else, so having the same inline is efficient.